subsys/bluetooth/audio/mics.c

```diff
@@ -0,0 +1,502 @@
+/*  Bluetooth MICS
+ *
+ * Copyright (c) 2020 Bose Corporation
+ * Copyright (c) 2020-2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/byteorder.h>
+#include <sys/check.h>
+
+#include <device.h>
+#include <init.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/audio/mics.h>
+#include <bluetooth/audio/aics.h>
+
+#include "mics_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_MICS)
+#define LOG_MODULE_NAME bt_mics
+#include "common/log.h"
+
+#if defined(CONFIG_BT_MICS)
```

Comment:
mics.c is compiled for both MICS and MICS_CLIENT. Thus there are things here that are only relevant to MICS, but the file is still compiled with MICS (for MICS_CLIENT). 

The part where we checked for either MICS || MICS_CLIENT shouldn't be there and was an oversight and has been removed. 

The third part (I assume the common API handlers?) are protected by various option to handle the various ways the function can be called for MICS=n, MICS=y, MICS_CLIENT=y and MICS_CLIENT=n. 

I think it would become a lot simpler if we had a common file for MICS and MICS_CLIENT, such that we could separate the common part from MICS and MICS_CLIENT. (We can do that, but should also be done to VCS, AICS and VOCS then).